src/main/java/com/aws/greengrass/builtin/services/pubsub/SubscriptionTrie.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.builtin.services.pubsub;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Trie to manage subscriptions.
+ */
+public class SubscriptionTrie {
+    private static final char TOPIC_LEVEL_SEPARATOR = '/';
+    private static final char SINGLE_LEVEL_WILDCARD = '+';
+    private static final char MULTI_LEVEL_WILDCARD = '#';
+
+    private final Map<String, SubscriptionTrie> children = new ConcurrentHashMap<>();
+    @SuppressWarnings("PMD.UnusedPrivateField")
+    @SuppressFBWarnings("URF_UNREAD_FIELD")
+    private final String value;
```

Comment:
Why is `value` field unused? If `callbacks` is basically the "value" for the trie node, can we remove this field?